fe/fe-core/src/main/java/com/starrocks/planner/OlapTableSink.java

```diff
@@ -429,8 +430,17 @@ private TOlapTableLocationParam createLocation(OlapTable table) throws UserExcep
             for (MaterializedIndex index : partition.getMaterializedIndices(IndexExtState.ALL)) {
                 for (Tablet tablet : index.getTablets()) {
                     if (table.isLakeTable()) {
+                        Warehouse warehouse = GlobalStateMgr.getCurrentState().getWarehouseMgr().
+                                getWarehouse(ConnectContext.get().getCurrentWarehouse());
```

Comment:
Should use the warehouse in the load job and the warehouse may be specified by user